hs_app_timeseries/templates/pages/timeseriesresource.html

```diff
@@ -217,6 +244,303 @@ <h3 style="color: red" id="file-type-error"></h3>
                 $(el).attr('value', 'No')
             }
         }
+        function processSiteMetadataElement(){
+            var sites_text = $('#id_available_sites').val();
+            var sites_json = $.parseJSON(sites_text);
+            var original_site_code = $('#id_site_code').val();
+            // remove the select option "Other.." from site code selection dropdown
+            $("#id_site_code_choices option[value='Other']").remove();
+
+            $('#id_site_code_choices').change(function () {
+                var selectedSiteCode = this.value;
+                if(selectedSiteCode === "----"){
+                    $('#id-site').find("button.btn-primary").hide();
+                    $('#id_site_code').val("");
+                    $('#id_site_code').attr('readonly', false);
+                    $('#id_site_name').val("");
+                    $('#id_site_name').attr('readonly', false);
+                    $('#id_elevation_m').val("");
+                    $('#id_elevation_m').attr('readonly', false);
+                    $('#id_elevation_datum').val("");
+                    $('#id_elevation_datum').attr('readonly', false);
+                    $('#id_site_type').val("");
+                    $('#id_site_type').attr('readonly', false);
+                    $('#id_latitude').val("");
+                    $('#id_latitude').attr('readonly', false);
+                    $('#id_longitude').val("");
+                    $('#id_longitude').attr('readonly', false);
```

Comment:
There should be a way to put all these ```#id``` elements into an array and bulk update their attributes to set ```readonly``` to false all together rather than doing them one by one.